### PR TITLE
Iteration 2 GUI

### LINKED ISSUE #9: Login Window, logging in after error.
Unable to login after a failed attempt.

### LINKED ISSUE #12: Register Screen, registering with non alphanumeric characters.
It is possible to register with usernames such as 'Daniel 3' and any email address format even without the '@' symbol.

### LINKED ISSUE #15: Same JavaFX Stage for every new Scene
A new stage is opened for every scene. Should just change Scene and Controller.

### PR DESCRIPTION
**Pull request by JADGardner**
### TP003 - Graphical User Interface
**User Stories Complete:** 031
**Authors:** James Gardner and Stijn Marynissen

Closes #9 
Closes #12 
Closes #15

### CODE REVIEW

File: client/src/main/java/application/controller/RegisterWindowController.java [hunk 1]
```diff
@@ -165,22 +166,74 @@ private boolean fieldsAreValid() {
...
+    if (specialCharPatten.matcher(username).find() || whiteSpace.matcher(username).find()
+    || digitCasePatten.matcher(username).find()) {
+      errorLabel.setText("Username should only contains letters and have no spaces");
+      return false;
+    }
+    return true;
+  }
+
+  protected Boolean emailIsValid(String email) {
+    String regex = "^[\\w-_\\.+]*[\\w-_\\.]\\@([\\w]+\\.)+[\\w]+[\\w]$";
```
Comment (cmckirgan) [inline client/src/main/java/application/controller/RegisterWindowController.java]: This is technically too strict, there are some weird emails it'll reject. Do we care?

File: client/src/main/java/application/view/ViewInitialiser.java [hunk 2]
```diff
@@ -12,29 +12,50 @@ public ViewInitialiser() {
 
   }
 
+//  /** Public method used to initialise stages from
```
Comment (cmckirgan) [inline client/src/main/java/application/view/ViewInitialiser.java]: Can you tidy this up or do you want to keep that?

Comment (JADGardner) [reply client/src/main/java/application/view/ViewInitialiser.java]: Fixed lad 👍

File: client/src/main/java/application/controller/RegisterWindowController.java (hunk 1 above)
Comment (JADGardner) [reply client/src/main/java/application/controller/RegisterWindowController.java]: Do you have a regex that would be less strict you think might be better?

Comment (cmckirgan) [reply client/src/main/java/application/controller/RegisterWindowController.java]: Email is a fun one to validate, https://haacked.com/archive/2007/08/21/i-knew-how-to-validate-an-email-address-until-i.aspx/ This is a decent write up tho the one he comes up with is a bit wrong due to his understanding of the standard

Comment (JADGardner) [reply client/src/main/java/application/controller/RegisterWindowController.java]: I've changed the regex to:

^[\\w!#$%&'*+/=?`{|}~^-]+(?:\\.[\\w!#$%&'*+/=?`{|}~^-]+)*@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,6}$

And we can catch anymore with an email confirmation, which were thinking of implementing anyway for checking tutors.

**Review by cmckirgan — APPROVED** [review]
Looks good. Approved

**Review by dmb537 — APPROVED** [review]
Looks good

**Pull request merged**